core/flow/rule_manager.go

```diff
@@ -3,42 +3,154 @@ package flow
 import (
 	"encoding/json"
 	"fmt"
+	"sync"
+
 	"github.com/alibaba/sentinel-golang/logging"
 	"github.com/pkg/errors"
-	"sync"
 )
 
-// const
 var (
 	logger = logging.GetDefaultLogger()
 )
 
+// SafeTrafficControllers represents the threadSafe map storage for TrafficShapingController.
+type SafeTrafficControllers struct {
+	controller map[string][]*TrafficShapingController
+	mux        sync.Mutex
+}
+
+// NewSafeTrafficControllers returns a new SafeTrafficControllers instance.
+func NewSafeTrafficControllers() *SafeTrafficControllers {
```

Comment:
Expose the function "NewSafeTrafficControllers" is not good practice.

Reply:
but "SafeTrafficControllers" itself is exposed...